src/openvpn/xkey_helper.c

```diff
@@ -205,4 +216,123 @@ xkey_management_sign(void *unused, unsigned char *sig, size_t *siglen,
     return (*siglen > 0);
 }
 
+/**
+ * Add PKCS1 DigestInfo to tbs and return the result in *enc.
+ *
+ * @param enc           pointer to output buffer
+ * @param enc_len       capacity in bytes of output buffer
+ * @param mdname        name of the hash algorithm (SHA256, SHA1 etc.)
+ * @param tbs           pointer to digest to be encoded
+ * @param tbslen        length of data in bytes
+ *
+ * @return              false on error, true  on success
+ *
+ * On return enc_len is  set to actual size of the result.
+ * enc is NULL or enc_len is not enough to store the result, it is set
+ * to the required size and false is returned.
+ */
+bool
+encode_pkcs1(unsigned char *enc, size_t *enc_len, const char *mdname,
```

Comment:
This function is really ugly but I also don't see a better way if other libraries really require of to add that stuff

Reply:
Ugly, yes.. but looks unavoidable to me.
For `--management-external-key` management-notes.txt says `RSA_private_encrypt()` will generate the right signature, which means digest-info header should be added by us. That said, the doc is probably from times when only MD5-SHA1 signature existed and there is no header needed for that. In mbedTLS code-path we manually add it before sending the digest to the management client.

For Windows it may be possible to convince it to do add the header while signing -- I haven't checked.

For pkcs11, I do not see any way to get it added by the token.